alica_engine/include/engine/BasicPlan.h

```diff
@@ -26,6 +28,7 @@ class BasicPlan : private RunnableObject
     virtual void onInit(){};
     virtual void run(void* msg){};
     virtual void onTerminate(){};
+    virtual std::unordered_map<int64_t, std::unordered_set> getApplicationEntrypointContext(vector<int64_t> entryPointIds){};
```

Comment:
- std::vector
- Can return const reference. 
- This default implementation will fail to compile. You can make it return a default constructed value.

Reply:
Add a comment explaining the inputs and outputs.